txscript/engine.go

```diff
@@ -411,20 +427,22 @@ func checkMinimalDataPush(op *opcode, data []byte) error {
 // whether or not it is hidden by conditionals, but some rules still must be
 // tested in this case.
 func (vm *Engine) executeOpcode(op *opcode, data []byte) error {
-	// Disabled opcodes are fail on program counter.
-	if isOpcodeDisabled(op.value) {
+	// Disabled opcodes are fail on program counter, but only if we're not
+	// executing a taproot script.
+	if vm.taprootCtx == nil && isOpcodeDisabled(op.value) {
 		str := fmt.Sprintf("attempt to execute disabled opcode %s", op.name)
 		return scriptError(ErrDisabledOpcode, str)
 	}
 
-	// Always-illegal opcodes are fail on program counter.
-	if isOpcodeAlwaysIllegal(op.value) {
+	// Always-illegal opcodes are fail on program counter, but only if
+	// we're not executing a taproot script.
+	if vm.taprootCtx == nil && isOpcodeAlwaysIllegal(op.value) {
```

Comment:
Why is there an exception for this with Taproot? The newly introduced success opcodes are all part of `isOpcodeDisabled`, so I can see why the exception exists there, but `OP_VERIF` and `OP_VERNOTIF` (illegal opcodes) are not part of that set.

Reply:
Hmm, yeah you're right that this is just a case for those two particular op codes. Reflecting, I think I added this here to make it uniform with the check about ti ognire disabled op codes. Re the success op codes, execution is short circuited elsewhere now (before we even step through any op codes), so this can be safely removed (the guard around is opcode disabled).